### PR TITLE
User credentials for influxdb user/password

### PR DESCRIPTION
**Pull request by jeffpearce**
Addresses #7 

Use credentials for influxdb user/password.

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig/config.jelly [hunk 1]
```diff
@@ -1,22 +1,27 @@
 <?jelly escape-by-default='true'?>
-<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:c="/lib/credentials">
```
Comment (tspanindra) [inline src/main/resources/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig/config.jelly]: I see this help text comes from a different html file. Nice!

File: src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig.java [hunk 2]
```diff
@@ -208,4 +284,96 @@ public void setInfluxDbRetentionPolicy(String influxDbRetentionPolicy) {
...
+        if (Strings.isNullOrEmpty(value)) {
+            return FormValidation.ok();
+        }
+        return FormValidation.warning("Using a plain text password is not secure - please use credentials instead");
+    }
+
+    /**
+     * Validates the credentialsId
+     * @param item context for validation
+     * @param value value to validate
```
Comment (tspanindra) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig.java]: nit: value repeated twice

File: src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig.java [hunk 3]
```diff
@@ -208,4 +284,96 @@ public void setInfluxDbRetentionPolicy(String influxDbRetentionPolicy) {
...
+        return FormValidation.ok();
+    }
+
+    private void loadCredentials() {
+        UsernamePasswordCredentials credentials
+                = getCredentials(UsernamePasswordCredentials.class,
+                        credentialsId);
+        secureDbUser = credentials.getUsername();
+        secureDbPassword = credentials.getPassword().getPlainText();
+
```
Comment (tspanindra) [inline src/main/java/org/jenkinsci/plugins/githubautostatus/BuildStatusConfig.java]: nit: extra line. not imp!

**Comment by oleg-nenashev** [comment]
I added it to my queue, but I am not sure I will get to it this week. Please do not consider my review as blocker

**Comment by jeffpearce** [comment]
@oleg-nenashev whenever you can get to it is fine. The main area I wanted to get outside feedback for is the usage of credentials, eg. https://github.com/jenkinsci/github-autostatus-plugin/pull/8/files#diff-d78d7a21a63edb0437c0b3e06fcbaa07R293 and https://github.com/jenkinsci/github-autostatus-plugin/pull/8/files#diff-d78d7a21a63edb0437c0b3e06fcbaa07R341. Comments are welcome on any part of the code if you have the time though.

**Pull request merged**